packages/jupyterlab-lsp/src/features/highlights.ts

```diff
@@ -145,6 +145,11 @@ export class HighlightsCM extends CodeMirrorIntegration {
       return;
     }
 
+    if (!root_position) {
```

Comment:
Would the following work as well for you?

```suggestion
    if (root_position == null) {
```

If yes, then I would slightly prefer it for consistency with other places in codebase.

Reply:
That works for this particular issue, but getCursor may return undefined as well. For example: https://github.com/krassowski/jupyterlab-lsp/blob/d86925f4a1c51c5d0fe76c1d2be74b49ffb71927/packages/jupyterlab-lsp/src/virtual/codemirror_editor.ts#L66

Reply:
The magic of null is that it `null == undefined` (not that I propose == rather than ===)

<img src="https://i.stack.imgur.com/nkpj6.png" width=400>

Reply:
To elaborate `!position` could bite in the future, should we decide to represent positions as integer offsets (I do not really believe that it would happen, but it is used by some editors like that and it might happen).

Reply:
and then `!0` would give `true`

Reply:
Oh, yea, missed the == vs ===. That change seems fine.

Reply:
Thanks for the clarification!